src/goMain.ts

```diff
@@ -289,6 +290,30 @@ export function activate(ctx: vscode.ExtensionContext): void {
 
 	ctx.subscriptions.push(vscode.commands.registerCommand('go.get.package', goGetPackage));
 
+	ctx.subscriptions.push(vscode.commands.registerCommand('go.lint.package', () => {
+		let editor = vscode.window.activeTextEditor;
+		let documentUri = editor ? editor.document.uri : null;
+		let goConfig = vscode.workspace.getConfiguration('go', documentUri);
+
+		lintCurrentPackage(documentUri, goConfig)
+			.then(handleDiagnosticErrors(editor ? editor.document : null))
```

Comment:
calling `handleDiagnosticErrors` clears any existing build and vet errors. It should only update the warnings.